userspace/libsinsp/tuples.cpp

```diff
@@ -23,6 +23,17 @@ limitations under the License.
 #include "sinsp_exception.h"
 #include "logger.h"
 
+
+/*
+ *  a few line below to enforce proper memory layout for ipv6addr class during compile time
+ */
+template <bool T> class mem_layout_test;
+template <> class mem_layout_test<true>{}; // empty struct take 1 byte of mem
+
+static auto check_size   = mem_layout_test<sizeof (ipv6addr) == 16>();
+static auto check_layout = mem_layout_test<std::is_pod<ipv6addr>::value>();
```

Comment:
Neat! I didn't know about is_pod().

Reply:
Yep, agree, very neat!  
Unfortunately it seems like `is_pod` is being deprecated in c++20 in favor of `std::is_standard_layout` and `std::is_trivial`; as doc says:
> If T is a [PODType](https://en.cppreference.com/w/cpp/named_req/PODType) ("plain old data type"), that is, both trivial and standard-layout, provides the member constant value equal to true. For any other type, value is false.

Therefore, i'd go with `is_standard_layout && is_trivial` if you're ok with it, small change to be future proof.  
I swear i will approve then :laughing: